x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java

```diff
@@ -1107,6 +1108,445 @@ public void testTypeConflictInObject() throws IOException {
         );
     }
 
+    /**
+     * https://github.com/elastic/elasticsearch/issues/117054
+     * One index with (Responses.process is nested; process.parent.command_line is supported):
+     * <pre>
+     *         "Responses": {
+     *           "properties": {
+     *             "process": {
+     *               "type": "nested",
+     *               "properties": {
+     *                 "pid": {
+     *                   "type": "long"
+     *                 }
+     *               }
+     *             }
+     *           }
+     *         },
+     *         "process": {
+     *           "properties": {
+     *             "parent": {
+     *               "properties": {
+     *                 "command_line": {
+     *                   "type": "wildcard",
+     *                   "fields": {
+     *                     "text": {
+     *                       "type": "text"
+     *                     }
+     *                   }
+     *                 }
+     *               }
+     *             }
+     *           }
+     *         }
+     * </pre>.
+     */
+    public void testOneNestedSubField_AndSameNameSupportedField() throws IOException {
+        ESRestTestCase.createIndex("test", Settings.EMPTY, """
+            "properties": {
+              "Responses": {
+                "properties": {
+                  "process": {
+                    "type": "nested",
+                    "properties": {
+                      "pid": {
+                        "type": "long"
+                      }
+                    }
+                  }
+                }
+              },
+              "process": {
+                "properties": {
+                  "parent": {
+                    "properties": {
+                      "command_line": {
+                        "type": "wildcard",
+                        "fields": {
+                          "text": {
+                            "type": "text"
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+            """);
+        index("test", """
+            {"Responses.process.pid": 123,"process.parent.command_line":"run.bat"}""");
+
+        Map<String, Object> result = runEsql("FROM test");
```

Comment:
Versions w/o this fix fail already at querying an empty index[*]. So a (totally optional) suggestion would be run a `FROM test` before indexing.

[*] On an 8.16:
```
java.lang.IllegalStateException: can't find input for [<no-fields>{r$}#26]
	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.planProject(LocalExecutionPlanner.java:602)
	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:203)
	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.planExchangeSink(LocalExecutionPlanner.java:321)
	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:235)
	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:174)
	at org.elasticsearch.xpack.esql.plugin.ComputeService.runCompute(ComputeService.java:446)
```

Reply:
I've added the additional test. For the `no-fields` issue, likely it has something to do with an empty mapping. See https://github.com/elastic/elasticsearch/issues/111545 where "empty mapping" has slightly different meanings depending on how the index is created and how field_caps behaves.